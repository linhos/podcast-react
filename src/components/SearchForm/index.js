import React, {useState} from 'react';
import {useHistory} from 'react-router-dom';

const SearchForm = () => {
    const history = useHistory();

    const [search, setSearch] = useState('');

    function handleChange(e) {
        setSearch(e.target.value);
    }

    function handleSubmit(event){
        event.preventDefault()
        history.push(`/search/${search}`);
    }

    return (
        <form onSubmit={handleSubmit}>
            <label >Buscar</label>
            <input type="text" onChange={handleChange} />
        </form>
    )

}

export {SearchForm};