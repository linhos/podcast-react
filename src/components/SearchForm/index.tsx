import React, {useState} from 'react';
import {useHistory} from 'react-router-dom';

const SearchForm: React.FC = () => {
    const history = useHistory();

    const [search, setSearch] = useState('');

    function handleChange(e: React.ChangeEvent<HTMLInputElement>) {
        const value:string = e.target.value;
        setSearch(value);
    }

    function handleSubmit(e: React.FormEvent<HTMLFormElement>){
        e.preventDefault()
        history.push(`/search/favourites`);
    }

    return (
        <form onSubmit={handleSubmit}>
            <label >Buscar</label>
            <input type="text" onChange={handleChange} placeholder="que quieres buscar?" />
        </form>
    )

}

export {SearchForm};