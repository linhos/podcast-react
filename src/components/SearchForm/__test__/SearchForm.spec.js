import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import { SearchForm } from "../../SearchForm";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

test('renders learn react link', () => {
    const history = createMemoryHistory();
    const { getByText } = render(
            <Router history={history}>
                <SearchForm />
            </Router>
    );
    
    const linkElement = getByText(/Buscar/i);
    expect(linkElement).toBeInTheDocument();
});

it('renders', () => {
    const history = createMemoryHistory();
    const {asFragment} = render(
        <Router history={history}>
                <SearchForm />
            </Router>
    )
    expect(asFragment()).toMatchSnapshot();
})
