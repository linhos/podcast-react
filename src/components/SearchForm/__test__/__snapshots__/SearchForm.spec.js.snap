// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <form>
    <label>
      Buscar
    </label>
    <input
      type="text"
    />
  </form>
</DocumentFragment>
`;
